File: Math/Max Points on a Line/MaxPoints.cs
public class Solution {
    public int MaxPoints(int[][] points) {
        int counter = 1;
        if (points.Length < 2) {
            return 1;
        }
        
        for (int i = 0; i < points.Length; i++) {
            Dictionary<double, int> slope = new Dictionary<double, int>();
            for (int j = i + 1; j < points.Length; j++) {
                int y = points[j][1] - points[i][1];
                int x = points[j][0] - points[i][0];
                if (x != 0) {
                    double currentSlope = (double)y / x;
                    if (!slope.ContainsKey(currentSlope)) {
                        slope[currentSlope] = 0;
                    }
                    slope[currentSlope]++;
                } else {
                    if (!slope.ContainsKey(double.PositiveInfinity)) {
                        slope[double.PositiveInfinity] = 0;
                    }
                    slope[double.PositiveInfinity]++;
                }
            }
            foreach (var kvp in slope) {
                counter = Math.Max(counter, kvp.Value);
            }
        }
                
        return counter + 1;
    }
}

File: Math/Max Points on a Line/MaxPoints.go
func maxPoints(points [][]int) int {
    counter := 1
	if len(points) < 2 {
		return 1
	}

	for i := 0; i < len(points); i++ {
		slope := make(map[float64]int)
		for j := i + 1; j < len(points); j++ {
			y := points[j][1] - points[i][1]
			x := points[j][0] - points[i][0]
			if x != 0 {
				currentSlope := float64(y) / float64(x)
				slope[currentSlope] = 1 + slope[currentSlope]
			} else {
				slope[math.Inf(1)] = 1 + slope[math.Inf(1)]
			}
		}
		for _, value := range slope {
			if value > counter {
				counter = value
			}
		}
	}

	return counter + 1
}